src/vs/workbench/contrib/chat/browser/chatInputPart.ts

```diff
@@ -429,17 +429,23 @@ export class ChatInputPart extends Disposable implements IHistoryNavigationWidge
 		return `chat.currentLanguageModel.${this.location}`;
 	}
 
+	private getSelectedModelIsDefaultStorageKey(): string {
+		return `chat.currentLanguageModel.${this.location}.isDefault`;
+	}
+
 	private initSelectedModel() {
 		const persistedSelection = this.storageService.get(this.getSelectedModelStorageKey(), StorageScope.APPLICATION);
+		const persistedIsDefault = this.storageService.getBoolean(this.getSelectedModelIsDefaultStorageKey(), StorageScope.APPLICATION, persistedSelection === 'github.copilot-chat/gpt-4o');
+
 		if (persistedSelection) {
 			const model = this.languageModelsService.lookupLanguageModel(persistedSelection);
-			if (model) {
+			if (model && (!model.isDefault || !persistedIsDefault)) {
 				this.setCurrentLanguageModel({ metadata: model, identifier: persistedSelection });
 				this.checkModelSupported();
 			} else {
 				this._waitForPersistedLanguageModel.value = this.languageModelsService.onDidChangeLanguageModels(e => {
 					const persistedModel = e.added?.find(m => m.identifier === persistedSelection);
-					if (persistedModel) {
+					if (persistedModel && (!persistedModel.metadata.isDefault || !persistedIsDefault)) {
```

Comment:
If we have `persistedModel`, do `this._waitForPersistedLanguageModel.clear();` so you don't keep waiting

Reply:
Wait shouldn't it be `persistedModel.metadata.isDefault`, no `!`?